crates/goose/src/providers/base.rs

```diff
@@ -381,6 +363,29 @@ pub trait Provider: Send + Sync {
 
         Ok(sanitized_description)
     }
+
+    // Generate a prompt for a session name based on the conversation history
+    fn create_session_name_prompt(&self, messages: &[Message]) -> String {
+        // Create a prompt for a concise description
+        let mut prompt = "Based on the conversation so far, provide a concise description of this session in 4 words or less. This will be used for finding the session later in a UI with limited space - reply *ONLY* with the description".to_string();
+
+        // Get context from the first 3 user messages
+        let context: Vec<String> = messages
+            .iter()
+            .filter(|m| m.role == rmcp::model::Role::User)
+            .take(3)
+            .map(|m| m.as_concat_text())
+            .collect();
```

Comment:
should we pass this in, instead of messages? the outer callers are only going to call us with the first three messages anyway (which I guess also means we should probably not cut at 3 here at all)